modules/service/src/main/scala/lucuma/odb/api/service/Connection.scala

```diff
@@ -151,17 +169,16 @@ object Connection {
     subscriptions:     Subscriptions[F]
   )(implicit F: ConcurrentEffect[F]) extends ConnectionState[F] {
 
-    override def init(
+    override def reset(
       u: Option[User],
       r: Option[FromServer] => F[Unit],
       s: Subscriptions[F]
     ): (ConnectionState[F], F[Unit]) =
-      terminate.map { action =>
-        for {
-          _ <- info(u, "received connection_init on already initialized connection")
-          _ <- action
-        } yield ()
-      }
+      (new Connected(odbService, u, r, s),
+        subscriptions.removeAll  *>
+          r(Some(ConnectionAck)) *>
+          r(Some(ConnectionKeepAlive))
+      )
```

Comment:
This is the big change.  A reset on an active connection just removes existing subscriptions and keeps on going.  There's no need to send a `TerminateConnection`, though you can if you want.

Reply:
If I understand correctly, `connection_init` now just resets subscriptions (and replies `ack` + `ka`). What happens if the client sends a `start` before calling `connection_init`? (or between `connection_terminate` and `connection_init`)

Reply:
If you call `start` before the first `connection_init` it will terminate the connection.  From then on a `connection_terminate` will just remove all subscriptions.  So a terminate followed by a `start` will leave you with one subscription.  Follow that will `connection_init` and it will remove the one subscription and update the user.

So really, there's no need to call `connection_terminate` unless you want to remove all subscriptions.